schema/omega-services.async.api.yaml

```diff
@@ -1,525 +1,415 @@
-      title: Update Result Status
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    AuthenticationError:
+      messageId: OELIFERR004
+      name: AuthenticationError
+      title: Authentication Error
+      description: >-
+        Authentication failed, likely due to an invalid token being present in
+        the `OMGToken` header.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/updateResponseStatus"
-      correlationId:
-        description: Default Correlation ID
-        location: $message.header#/correlationId
-
-    errorResponse:
-      name: errorResponse
-      title: Error response message
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    UnrecognisedMessageTypeError:
+      messageId: OELIFERR005
+      name: UnrecognisedMessageTypeError
+      title: Unrecognised Message Type Error
+      description: >-
+        The value of `OMGMessageTypeID` header was not recognised and so the
+        message cannot be further dispatched for processing.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
+      payload:
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    InvalidMessageError:
+      messageId: OELIFERR006
+      name: InvalidMessageError
+      title: Invalid Message Error
+      description: >-
+        The content of the body of the message has failed validation, e.g. a
+        parsed JSON message may be missing some required properties, or those
+        properties may have invalid values.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
+      payload:
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    ProcessingError:
+      messageId: OELIFERR007
+      name: ProcessingError
+      title: ProcessingError
+      description: >-
+        The received message was valid, but the system was unable complete the
+        processing of a message.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/errorMessage"
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+  messageTraits:
+    JMSMessage:
+      name: JMS Message
+      description: Headers that are mostly common to all JMS Messages
+      headers:
+        required:
+          - JMSMessageID
+          - JMSTimestamp
```

Comment:
These look like JMS-specific headers to me in which case they should not appear here (see https://www.asyncapi.com/docs/reference/specification/v2.6.0#messageObject). Rather they should appear in a bindings field something like this:
```
bindings:
    jms:
        headers:
            required:
                - JMSMessageID
                - JMSTimestamp
```
We need to make a clear distinction between the headers that are protocol specific and those that are application specific.

Reply:
I agree that there are no JMS-specific bindings defined, however the specification also states [here](https://www.asyncapi.com/docs/reference/specification/v2.6.0#messageObject) that the message object headers field is a..
>  Schema definition of the application headers. Schema MUST be of type "object". It MUST NOT define the protocol headers.
I would say these are protocol headers for JMS and if I move them into a `bindings.jms` section as follows, it is still valid and renders nicely:
```
messageTraits:
    JMSMessage:
      name: JMS Message
      description: Headers that are mostly common to all JMS Messages
      bindings:
        jms:
          headers:
            JMSMessageID:
              name: JMSMessageID
              description: >-
                A unique identifier for the message. We suggest using a short
                application instance specific prefix followed by a unique
                identifier. In the extreme case a UUID may be used.
              type: string
            JMSTimestamp:
              name: JMSTimestamp
              description: >-
                The time the message was sent. The value of the timestamp is the
                amount of time, measured in milliseconds, that has elapsed since
                midnight, January 1, 1970, UTC.
              type: number
```

Reply:
@rwalpole I have opened an issue with AsyncAPI to gain guidance on this: https://github.com/asyncapi/spec/issues/919

Reply:
@rwalpole They confirmed that at present there are no JMS Object bindings. So I have also sent them a PR to add such bindings to AsyncAPI - https://github.com/asyncapi/bindings/pull/193. It would be great if you could take a look at the asyncapi PR ASAP and add a comment to lend your support for it as well...

Reply:
@rwalpole So I think we could move forward here by leaving it as is for now, and then updating it when/if we can get the necessary changes made ti AsyncAPI.

Reply:
Agree. That makes sense.